Build failure: driftnet

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```
nix-shell -p driftnet
```

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
--------------------------------------------------
Running phase: buildPhase
build flags: -j12 SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
Making all in src
make[1]: Entering directory '/build/source/src'
Making all in common
make[2]: Entering directory '/build/source/src/common'
  CC       log.o
  CC       tmpdir.o
  CC       util.o
  AR       libcommon.a
make[2]: Leaving directory '/build/source/src/common'
Making all in network
make[2]: Entering directory '/build/source/src/network'
  CC       connection.o
  CC       layer2.o
  CC       layer3.o
  CC       pcap_engine.o
pcap_engine.c: In function 'network_get_default_interface':
pcap_engine.c:212:5: warning: 'pcap_lookupdev' is deprecated: use 'pcap_findalldevs' and use the first device [-Wdeprecated-declarations]
  212 |     interface = pcap_lookupdev(ebuf);
      |     ^~~~~~~~~
In file included from /nix/store/6hw4aym42f8xfnr6qgn5yss2l15rdn52-libpcap-1.10.5/include/pcap.h:43,
                 from pcap_engine.c:35:
/nix/store/6hw4aym42f8xfnr6qgn5yss2l15rdn52-libpcap-1.10.5/include/pcap/pcap.h:444:18: note: declared here
  444 | PCAP_API char   *pcap_lookupdev(char *);
      |                  ^~~~~~~~~~~~~~
  AR       libnetwork.a
make[2]: Leaving directory '/build/source/src/network'
Making all in media
make[2]: Entering directory '/build/source/src/media'
  CC       media.o
  CC       image.o
  CC       audio.o
  CC       mpeghdr.o
  CC       playaudio.o
  CC       http.o
playaudio.c: In function 'do_mpeg_player':
playaudio.c:233:5: warning: ignoring return value of 'pipe' declared with attribute 'warn_unused_result' [-Wunused-result]
  233 |     pipe(pp);
      |     ^~~~~~~~
  AR       libmedia.a
make[2]: Leaving directory '/build/source/src/media'
Making all in display
make[2]: Entering directory '/build/source/src/display'
  CC       display.o
  CC       img.o
  CC       gif.o
  CC       jpeg.o
  CC       png.o
  CC       webp.o
display.c: In function 'update_window':
display.c:153:9: warning: 'gdk_cairo_create' is deprecated: Use 'gdk_window_begin_draw_frame() and gdk_drawing_context_get_cairo_context()' instead [-Wdeprecated-declarations]
  153 |         cairo_t *cr = gdk_cairo_create(drawable);
      |         ^~~~~~~
In file included from /nix/store/r9qybwam7v4sj4b6jyavb7dm97hhprm9-gtk+3-3.24.51-dev/include/gtk-3.0/gdk/gdk.h:33,
                 from /nix/store/r9qybwam7v4sj4b6jyavb7dm97hhprm9-gtk+3-3.24.51-dev/include/gtk-3.0/gtk/gtk.h:30,
                 from display.c:23:
/nix/store/r9qybwam7v4sj4b6jyavb7dm97hhprm9-gtk+3-3.24.51-dev/include/gtk-3.0/gdk/gdkcairo.h:35:12: note: declared here
   35 | cairo_t  * gdk_cairo_create             (GdkWindow          *window);
      |            ^~~~~~~~~~~~~~~~
display.c: In function 'button_release_event':
display.c:323:9: warning: 'gdk_cairo_create' is deprecated: Use 'gdk_window_begin_draw_frame() and gdk_drawing_context_get_cairo_context()' instead [-Wdeprecated-declarations]
  323 |         cairo_t *cr = gdk_cairo_create(drawable);
      |         ^~~~~~~
/nix/store/r9qybwam7v4sj4b6jyavb7dm97hhprm9-gtk+3-3.24.51-dev/include/gtk-3.0/gdk/gdkcairo.h:35:12: note: declared here
   35 | cairo_t  * gdk_cairo_create             (GdkWindow          *window);
      |            ^~~~~~~~~~~~~~~~
display.c:330:9: warning: 'gdk_flush' is deprecated: Use 'gdk_display_flush' instead [-Wdeprecated-declarations]
  330 |         gdk_flush();    /* force X to actually draw the damn thing. */
      |         ^~~~~~~~~
In file included from /nix/store/r9qybwam7v4sj4b6jyavb7dm97hhprm9-gtk+3-3.24.51-dev/include/gtk-3.0/gdk/gdk.h:50:
/nix/store/r9qybwam7v4sj4b6jyavb7dm97hhprm9-gtk+3-3.24.51-dev/include/gtk-3.0/gdk/gdkmain.h:124:6: note: declared here
  124 | void gdk_flush (void);
      |      ^~~~~~~~~
display.c: In function 'display_send_img':
display.c:64:5: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
   64 |     write(imgpipe_writefd, name, len);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
display.c: In function 'do_image_display':
display.c:73:5: warning: ignoring return value of 'pipe' declared with attribute 'warn_unused_result' [-Wunused-result]
   73 |     pipe(pfd);
      |     ^~~~~~~~~
display.c: In function 'pipe_event':
display.c:420:29: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
  420 |                             write(1, "\a", 1);
      |                             ^~~~~~~~~~~~~~~~~
  AR       libdisplay.a
make[2]: Leaving directory '/build/source/src/display'
Making all in http_display
make[2]: Entering directory '/build/source/src/http_display'
rm -f 'web_data.c'
  CC       httpd.o
echo '#include "web_data.h"' >> web_data.c
echo 'char* static_files[] = {' >> web_data.c
for f in static_web/index.html ; do \
        basename $f | sed 's@\(.*\)@"\1",@g' >> web_data.c; \
done
echo '0};' >> web_data.c
  CC       web_data.o
httpd.c: In function 'http_server_dispatch':
httpd.c:194:9: warning: unsigned conversion from 'int' to 'unsigned char:1' changes value from '2' to '0' [-Woverflow]
  194 |         LWSMPRO_FILE,                   /* mount type is a directory in a filesystem */
      |         ^~~~~~~~~~~~
httpd.c:196:9: error: initialization of 'unsigned char' from 'void *' makes integer from pointer without a cast [-Wint-conversion]
  196 |         NULL,
      |         ^~~~
httpd.c:196:9: note: (near initialization for 'mount.mountpoint_len')
make[2]: *** [Makefile:382: httpd.o] Error 1
make[2]: Leaving directory '/build/source/src/http_display'
make[1]: *** [Makefile:463: all-recursive] Error 1
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:435: all-recursive] Error 1
error: Cannot build '/nix/store/3ndsrm4allx393ghbmd0y5f3fhg8akgw-driftnet-1.6.0.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/1zadpvkxxxn4sb1kkxaayrq99ikx88jy-driftnet-1.6.0
```

### Additional context

_No response_

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.60-1-lts, Athena OS, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.32.4`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/l0ch8xv4zjwmzkvxir8g9g9vq3f6dhdc-nixpkgs/nixpkgs`
```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)
@offlinehacker 

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/dr/driftnet/package.nix
+++ b/pkgs/by-name/dr/driftnet/package.nix
@@ -49,6 +49,8 @@ stdenv.mkDerivation rec {
     openssl
   ];
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=int-conversion";
+
   meta = {
     description = "Watches network traffic, and picks out and displays JPEG and GIF images for display";
     homepage = "https://github.com/deiv/driftnet";

